.github/workflows/miri-check.yml

```diff
@@ -0,0 +1,37 @@
+name: Miri
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main, release* ]
+
+jobs:
+  miri:
+    name: "Miri"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Miri
+        run: |
+          rustup toolchain install nightly --component miri
+          rustup override set nightly
+          cargo miri setup
+      - name: Test with Miri
+        run: |
+          # the script will be run at the project root, so use filename directly
```

Comment:
@xieyuschen Could you put this logic in a script in `internal/scripts` e.g. `ci_run_miri.sh`.

It should be possible for developers to execute the script in their own environment to reproduce any CI issues. This would reduce some friction when addressing CI failures.

Reply:
done, i have verified it in my forked repo, [ref](https://github.com/xieyuschen/iceoryx2/pull/2)

Reply:
@xieyuschen 

Since we are an Eclipse open-source project, the script needs the following header:

```shell
# Copyright (c) 2024 Contributors to the Eclipse Foundation
#
# See the NOTICE file(s) distributed with this work for additional
# information regarding copyright ownership.
#
# This program and the accompanying materials are made available under the
# terms of the Apache Software License 2.0 which is available at
# https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
# which is available at https://opensource.org/licenses/MIT.
#
# SPDX-License-Identifier: Apache-2.0 OR MIT
```

Sorry for the hassle, it is tedious and easy to forget 😓  (hence the CI job).

With that done, it should be fine to merge.

Reply:
oh, nvm let me add it. btw, I have saw the contribution docs and it says to add the copyright for _source code_ only.

let me add it in both `.sh` and new ci file.

Reply:
@xieyuschen You can skip the CI file :)

Reply:
done, just added the license comment for the `.sh`